metal/end_model/loss.py

```diff
@@ -20,14 +20,19 @@ class SoftCrossEntropyLoss(nn.Module):
 
     def __init__(self, weight=None, reduction="elementwise_mean"):
         super().__init__()
-        assert weight is None or isinstance(weight, torch.FloatTensor)
         self.weight = weight
+        if self.weight is not None:
+            self.weight = torch.Tensor(self.weight)
+        assert self.weight is None or isinstance(self.weight, torch.FloatTensor)
         self.reduction = reduction
 
     def forward(self, input, target):
         n, k = input.shape
         # Note that t.new_zeros, t.new_full put tensor on same device as t
         cum_losses = input.new_zeros(n)
+        # Moving weights to appropriate device
+        if self.weight is not None:
+            self.weight = self.weight.type(input.dtype).to(input.device)
```

Comment:
This seems clunky?  It seems like the torch-preferred way is to use [`register_buffer`](https://pytorch.org/docs/stable/nn.html#torch.nn.Module.register_buffer) here